derive/src/max_encoded_len.rs

```diff
@@ -0,0 +1,203 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use proc_macro_crate::{crate_name, FoundCrate};
+use proc_macro2::{Span, Ident};
+use quote::{quote, quote_spanned};
+use syn::{
+	Data, DeriveInput, Error, Fields, GenericParam, Generics, Meta, TraitBound, Type, TypeParamBound,
+	parse_quote, spanned::Spanned,
+};
+
+/// impl for `#[derive(MaxEncodedLen)]`
+pub fn derive_max_encoded_len(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+	let input: DeriveInput = match syn::parse(input) {
+		Ok(input) => input,
+		Err(e) => return e.to_compile_error().into(),
+	};
+
+	let mel_trait = match max_encoded_len_trait(&input) {
+		Ok(mel_trait) => mel_trait,
+		Err(e) => return e.to_compile_error().into(),
+	};
+
+	let name = &input.ident;
+	let generics = add_trait_bounds(input.generics, mel_trait.clone());
+	let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
+
+	let data_expr = data_length_expr(&input.data);
+
+	quote::quote!(
+		const _: () = {
+			impl #impl_generics #mel_trait for #name #ty_generics #where_clause {
+				fn max_encoded_len() -> usize {
+					#data_expr
+				}
+			}
+		};
+	)
+	.into()
+}
+
+fn max_encoded_len_trait(input: &DeriveInput) -> syn::Result<TraitBound> {
+	let mel = {
+		const EXPECT_LIST: &str = "expect: #[max_encoded_len_mod(path::to::crate)]";
+		const EXPECT_PATH: &str = "expect: path::to::crate";
+
+		macro_rules! return_err {
+			($wrong_style:expr, $err:expr) => {
+				return Err(Error::new($wrong_style.span(), $err))
+			};
+		}
+
+		let mut mel_crates = Vec::with_capacity(2);
+		mel_crates.extend(input
+			.attrs
+			.iter()
+			.filter(|attr| attr.path == parse_quote!(max_encoded_len_mod))
+			.take(2)
+			.map(|attr| {
+				let meta_list = match attr.parse_meta()? {
+					Meta::List(meta_list) => meta_list,
+					Meta::Path(wrong_style) => return_err!(wrong_style, EXPECT_LIST),
+					Meta::NameValue(wrong_style) => return_err!(wrong_style, EXPECT_LIST),
+				};
+				if meta_list.nested.len() != 1 {
+					return_err!(meta_list, "expected exactly 1 item");
+				}
+				let first_nested =
+					meta_list.nested.into_iter().next().expect("length checked above");
+				let meta = match first_nested {
+					syn::NestedMeta::Lit(l) => {
+						return_err!(l, "expected a path item, not a literal")
+					}
+					syn::NestedMeta::Meta(meta) => meta,
+				};
+				let path = match meta {
+					Meta::Path(path) => path,
+					Meta::List(ref wrong_style) => return_err!(wrong_style, EXPECT_PATH),
+					Meta::NameValue(ref wrong_style) => return_err!(wrong_style, EXPECT_PATH),
+				};
+				Ok(path)
+			})
+			.collect::<Result<Vec<_>, _>>()?);
+
+		// we have to return `Result<Ident, Error>` here in order to satisfy the trait
+		// bounds for `.or_else` for `generate_crate_access_2018`, even though `Option<Ident>`
+		// would be more natural in this circumstance.
```

Comment:
You are swallowing all errors below. So this makes no sense. Return an `Option<Result<>>` here to handle the errors.